src/translators/decode.rs

```diff
@@ -175,17 +177,27 @@ pub fn print_decoded_token(
         };
     }
 
-    match (format, token_data) {
-        (OutputFormat::Json, Ok(token)) => {
-            println!("{}", to_string_pretty(&TokenOutput::new(token)).unwrap())
+    match (output_path.as_ref(), format, token_data) {
+        (Some(path), _, Ok(token)) => {
+            let json = to_string_pretty(&TokenOutput::new(token)).unwrap();
+            write_file(path, json.as_bytes());
+            println!("Wrote jwt to file {}", path.display());
         }
-        (_, Ok(token)) => {
-            bunt::println!("\n{$bold}Token header\n------------{/$}");
-            println!("{}\n", to_string_pretty(&token.header).unwrap());
-            bunt::println!("{$bold}Token claims\n------------{/$}");
-            println!("{}", to_string_pretty(&token.claims).unwrap());
+        (None, OutputFormat::Json, Ok(token)) => {
+            println!("{}", to_string_pretty(&TokenOutput::new(token)).unwrap());
         }
-        (_, Err(_)) => exit(1),
+        (None, _, Ok(token)) => {
+            let header_output = to_string_pretty(&token.header).unwrap();
+            let claims_output = to_string_pretty(&token.claims).unwrap();
+            if output_path.is_some() {
+            } else {
```

Comment:
This seems like an unreachable if block, since the first match block would always run if `output_path` is `Some`, right?

Reply:
Thanks for spotting, looks like a remainder of a thought process ;). Fixed.